lib/ui/platform_dispatcher.dart

```diff
@@ -801,11 +801,36 @@ class PlatformDispatcher {
   ///
   ///  * [SchedulerBinding], the Flutter framework class which manages the
   ///    scheduling of frames.
+  ///  * [forceSyncFrame], a similar method that is used in rare cases that
+  ///    a frame must be rendered immediately.
   void scheduleFrame() => _scheduleFrame();
 
   @Native<Void Function()>(symbol: 'PlatformConfigurationNativeApi::ScheduleFrame')
   external static void _scheduleFrame();
 
+  /// Immediately render a frame by invoking the [onBeginFrame] and
+  /// [onDrawFrame] callbacks synchronously.
+  ///
+  /// This method performs the same computation for a frame as [scheduleFrame]
+  /// does, but instead of doing so at an appropriate opportunity, the render is
+  /// completed synchronously within this call.
+  ///
+  /// Prefer [scheduleFrame] to update the display in normal operation. The
+  /// [forceSyncFrame] method is designed for situations that require a frame is
+  /// rendered as soon as possible, even at the cost of rendering quality. An
+  /// example of using this method is [SchedulerBinding.scheduleWarmUpFrame],
+  /// which is called during application startup so that the first frame can be
+  /// presented to the screen a few extra milliseconds earlier.
+  ///
+  /// See also:
+  ///
+  ///  * [SchedulerBinding.scheduleWarmUpFrame], which uses this method.
+  ///  * [scheduleFrame].
```

Comment:
This can cause screen tearing & web doesn't support "out-of-vsync rendering".

Perhaps:

```suggestion
  /// Immediately render a frame by invoking the [onBeginFrame] and
  /// [onDrawFrame] callbacks synchronously.
  ///
  /// This forces a frame to be computed synchronously within this call,
  /// ignoring v-sync. Rendering in a forced frame may not be fully
  /// supported by all platforms.
  ///
  /// Prefer [scheduleFrame] to update the display in normal operation. The
  /// [forceSyncFrame] method is designed for situations that require a frame is
  /// rendered as soon as possible, even at the cost of rendering quality. An
  /// example of using this method is [SchedulerBinding.scheduleWarmUpFrame],
  /// which is called during application startup so that resources can be prepared
  /// before the first frame.
  ///
  /// See also:
  ///
  ///  * [SchedulerBinding.scheduleWarmUpFrame], which uses this method.
  ///  * [scheduleFrame].
```